### PR TITLE
[backend/frontend] Add constraint on inject's depends duration property

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Related issues

* https://github.com/OpenEx-Platform/openex/issues/345
* https://github.com/OpenEx-Platform/openex/issues/257

### CODE REVIEW

File: openex-api/src/main/java/io/openex/migration/V3_1__Inject_not_null_depends_duration.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.stereotype.Component;
+
+import java.sql.Connection;
+import java.sql.Statement;
+
+@Component
+public class V3_1__Inject_not_null_depends_duration extends BaseJavaMigration {
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/migration/V3_1__Inject_not_null_depends_duration.java:11]: Stick on V2 nomenclature

File: openex-api/src/main/java/io/openex/migration/V2_61__Inject_not_null_depends_duration.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.stereotype.Component;
+
+import java.sql.Connection;
+import java.sql.Statement;
+
+@Component
+public class V3_1__Inject_not_null_depends_duration extends BaseJavaMigration {
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/migration/V2_61__Inject_not_null_depends_duration.java:11]:
```suggestion
public class V2_61__Inject_not_null_depends_duration extends BaseJavaMigration {
```

File: openex-api/src/main/java/io/openex/rest/inject/form/InjectInput.java [hunk 3]
```diff
@@ -24,7 +28,11 @@ public class InjectInput {
     @JsonProperty("inject_depends_from_another")
     private String dependsOn;
 
+    @Getter
+    @Setter
     @JsonProperty("inject_depends_duration")
+    @NotNull
+    @Min(value = 0L, message = "The value must be positive")
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/rest/inject/form/InjectInput.java:35]:
```suggestion
```

File: openex-api/src/main/java/io/openex/rest/inject/form/InjectUpdateTriggerInput.java [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+@Getter
+@Setter
 public class InjectUpdateTriggerInput {
 
     @JsonProperty("inject_depends_duration")
+    @NotNull
+    @Min(value = 0L, message = "The value must be positive")
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/rest/inject/form/InjectUpdateTriggerInput.java:16]:
```suggestion
```

File: openex-api/src/main/java/io/openex/rest/inject/form/InjectUpdateTriggerInput.java [hunk 5]
```diff
@@ -1,17 +1,17 @@
 package io.openex.rest.inject.form;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.Setter;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/rest/inject/form/InjectUpdateTriggerInput.java:8]:
```suggestion
```

File: openex-api/src/main/java/io/openex/rest/inject/form/InjectInput.java [hunk 6]
```diff
@@ -3,7 +3,11 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import io.openex.database.model.Inject;
+import lombok.Getter;
+import lombok.Setter;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
```
Comment (RomuDeuxfois) [inline openex-api/src/main/java/io/openex/rest/inject/form/InjectInput.java:10]:
```suggestion
```

**Review by guillaumejparis — APPROVED** [review]

**Pull request merged**